### ISSUE TITLE
committing delete query throws UNKNOWN

### ISSUE THREAD

**Issue by sorsaffari** [graknlabs/grakn#4447]
Grakn version: 1.5.0-SNAPSHOT

## Expected output:
```
>>> match $pr isa protein has name "ORM1-like protein 3"; $sourcing ($pr) isa sourcing-of-information; delete $sourcing;
{V1032224, V73928}
>>> commit
>>>
```

## Actual output
```
>>> match $pr isa protein has name "ORM1-like protein 3"; $sourcing ($pr) isa sourcing-of-information; delete $sourcing;
{V1032224, V73928}
>>> commit
UNKNOWN: Vertex with id 1032224 was removed.. Please check server logs for the stack trace.
>>>
```

## Reproducing
```
git clone git@github.com:sorsaffari/biograkn.git
cd biograkn/blast
git checkout --track origin/blast-example
path/to/grakn-core-1.5.0-SNAPSHOT/graql console -k proteins -f schema.gql
pip3 install grakn
pip3 install biopython
python3 migrate.py
python3 blast.py
wait for blast output to be imported
enter 0 for all questions (cleaning data) > this is actually where the bug was identified. selecting a protein number from the list, throws the error in question.
path/to/grakn-core-1.5.0-SNAPSHOT/graql console -k proteins
>>> match $pr isa protein has name "ORM1-like protein 3"; $sourcing ($pr) isa sourcing-of-information; delete $sourcing;
{V1032224, V73928}
>>> commit
```

**Comment by sorsaffari** [comment]
@lolski just to give you some context on how the BLAST example(and its timeline) is associated with this bug: the part of the BLAST code that is blocked by this, is not the main part of the code and can be left out till this bug is fixed. 

We're planning to:
- publish the blog post and the example code on Friday (12th October). 
- present the work on BLAST <> Grakn in a meetup on (18th October).

If you've too much on your hand, it's completely fine. we don't _have to_ have that part of the BLAST example for these 2 milestones.

**Comment by BFergerson** [comment]
Just ran into this exact issue. Is there a workaround for this or do deletes just not work in Grakn 1.4.2?

**Comment by BFergerson** [comment]
@lolski, sure thing. Please check out this branch: https://github.com/bfergerson/grakn-calendar-test/tree/delete-test

Running it causes:
```
Ontology setup
Most recent thing id: [[$d/V16480][$m/V36880][$t/V28712]]
Most recent thing id: [[$d/V16480][$m/V20600][$t/V24688]]
Inserted things
Exception in thread "main" java.lang.RuntimeException: UNKNOWN: Vertex with id 36880 was removed.. Please check server logs for the stack trace.
	at ai.grakn.client.Grakn$Transaction.responseOrThrow(Grakn.java:218)
	at ai.grakn.client.Grakn$Transaction.commit(Grakn.java:228)
	at ai.grakn.GraknTx$commit$0.call(Unknown Source)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:45)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:108)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:112)
	at GraknCalendarTest.main(GraknCalendarTest.groovy:41)

Process finished with exit code 1
```

**Comment by lolski** [comment]
@BFergerson thanks, we will have a look at it.

**Comment by BFergerson** [comment]
@haikalpribadi, noticed you pushed this back to 1.6. Does that mean that delete queries won't in Grakn 1.5 as well?

**Comment by haikalpribadi** [comment]
I actually haven't had a chance to look into this, @BFergerson. All the unresolved issues from 1.5 milestone were moved to 1.6 in batch. Is this something critical in your current usage?

@flyingsilverfin can you help look into this issue?

**Comment by BFergerson** [comment]
@haikalpribadi, it would definitely be nice to have. I've been wanting to bring the next iteration of GitDetective and the new model (discussed it here https://discuss.grakn.ai/t/modeling-time-based-graphs-in-grakn/996) would require deleting relationships.

**Comment by haikalpribadi** [comment]
Alright, we'll look into it this week, @BFergerson ! :)

**Comment by flyingsilverfin** [comment]
So I've tried to reproduce @BFergerson nice small example - migrated to a relatively current version of 1.5 that we're close to releasing in a new repository (https://github.com/flyingsilverfin/grakn-debug/tree/master/delete-failure-brandon). I ran the test and it does not fail.

@BFergerson you can try to update to 1.5 a bit early and build it directly, or wait a week and confirm it fixes your bug, and if not we can take look at it again and try to fix it for 1.5.1?

**Comment by flyingsilverfin** [comment]
@sorsaffari is going to check if the delete bug from Biograkn also disappears when using a current 1.5 server as well :)

**Comment by flyingsilverfin** [comment]
@BFergerson seems like it isn't happening in BioGrakn on 1.5 either!

**Comment by BFergerson** [comment]
@flyingsilverfin, thanks for looking into this. I'm very glad to hear this has been fixed. Out of curiosity, how would I download an early build of Grakn 1.5? Is that the master branch?

**Comment by haikalpribadi** [comment]
Yes, @BFergerson. Just clone `master` and follow the build instruction here: https://github.com/graknlabs/grakn#compiling-grakn-core-from-source

**Comment by flyingsilverfin** [comment]
@BFergerson I've fixed the original BioGrakn issue, which stemmed from trying to delete inferred concepts - please open a new issue if you keep having issues!

**Comment by haikalpribadi** [comment]
Don't forget to move an issue to the correct milestone when you close it (and it wasn't originally in the correct one), @flyingsilverfin - thanks!
